### PR TITLE
Added MenuBar Item

### PR DESCRIPTION
**Pull request by sanarroya**
#2 

### Description

- Created MenuBarHandler which owns everything related to the status bar item and the menu options of it
- Added `.gitignore` file
- Added `CODEOWNERS`file
- Added Info.plist` entry so the app is not shown on the dock
- Added temporary icon for the status bar item

### CODE REVIEW

File: Bee/Application/MenuBar/MenuBarHandler.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    
+    override init() {
+        super.init()
+        buildStatusItem()
+    }
+}
+
+private extension MenuBarHandler {
+    
+    enum MenuItem: CaseIterable {
```
Comment (afterxleep) [inline Bee/Application/MenuBar/MenuBarHandler.swift:22]: Fuck yeah, case Iterable! 👍

File: Bee/Application/MenuBar/MenuBarHandler.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    
+    @objc func presentSettings() {
+        let contentView = ContentView()
+        let window = NSWindow(
+            contentRect: NSRect(x: 0, y: 0, width: 480, height: 300),
+            styleMask: [.titled, .closable, .miniaturizable, .resizable, .fullSizeContentView],
+            backing: .buffered, defer: false)
+        
+        guard let appDelegate = NSApplication.shared.delegate as? AppDelegate else {
```
Comment (afterxleep) [inline Bee/Application/MenuBar/MenuBarHandler.swift:53]: Shouldn't we pass this a dependency instead?

File: Bee/Application/MenuBar/MenuBarHandler.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                case .preferences:
+                    return #selector(presentSettings)
+                case .quit:
+                    return #selector(NSApplication.terminate(_:))
+            }
+        }
+        
+    }
+    
+    @objc func presentSettings() {
```
Comment (afterxleep) [inline Bee/Application/MenuBar/MenuBarHandler.swift:46]: Not sure how this works, but is this method overriding what the app delegate does at the start?

File: Bee/Application/MenuBar/MenuBarHandler.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    
+    var statusItem: NSStatusItem!
+    
+    override init() {
+        super.init()
+        buildStatusItem()
+    }
+}
+
+private extension MenuBarHandler {
```
Comment (afterxleep) [inline Bee/Application/MenuBar/MenuBarHandler.swift:20]: Whats the idea behind implementing everything under a private extension besides not writing private methods individually?

**Review by afterxleep — APPROVED** [review]
I do have some questions/comments, feel free to address at you consider necessary.

File: Bee/Application/MenuBar/MenuBarHandler.swift (hunk 2 above)
Comment (sanarroya) [reply Bee/Application/MenuBar/MenuBarHandler.swift:53]: Good catch

File: Bee/Application/MenuBar/MenuBarHandler.swift (hunk 3 above)
Comment (sanarroya) [reply Bee/Application/MenuBar/MenuBarHandler.swift:46]: If we leave this in the app delegate, then when the app is launched a screen will be presented to the user, this way the app will launch "minimized" on the status bar and the screen will be presented when the user taps on the preferences option of the menu.

Maybe we should present the preferences screen on first launch only? let me know what you think

File: Bee/Application/MenuBar/MenuBarHandler.swift (hunk 4 above)
Comment (sanarroya) [reply Bee/Application/MenuBar/MenuBarHandler.swift:20]: Just grouping them, is purely aesthetically, I prefer having private methods grouped and think an extension is a cool way to do it, but we can just put them together without an extension

File: Bee/Application/MenuBar/MenuBarHandler.swift (hunk 3 above)
Comment (afterxleep) [reply Bee/Application/MenuBar/MenuBarHandler.swift:46]: Yea, preferences on first launch is fine while we later on create an onboard tutorial or something.

File: Bee/Application/MenuBar/MenuBarHandler.swift (hunk 4 above)
Comment (afterxleep) [reply Bee/Application/MenuBar/MenuBarHandler.swift:20]: nah, looks good.

**Comment by sanarroya** [comment]
Created a new PR because I have some issues solving conflicts and wasted a lot of time

**Pull request closed without merging**